packages/mongodb-rag-core/src/index.ts

```diff
@@ -30,5 +29,5 @@ export * from "./assertEnvVars";
 
 // Everyone share the same versions of these packages
 export * from "mongodb";
-export * from "@azure/openai";
+export * as OpenAI from "openai";
 export * from "./langchain";
```

Comment:
@nlarew in future PR, i wanna do the hared langchain and mongodb exports w/ the `* as Name` convention for continuing the better namespacing of modules in this package,

ie:

```ts
export * as MongoDB from "mongodb";
export * as Langchain from "langchain";
```

not doing in this PR b/c it'd dramatically increase the # of files touched.

relatedly, i think it'd probably be *more correct* to change these external dependencies that we re-export as `peerDependencies` in the package.json, but also since our monorepo is the main consumer of the `mongodb-rag-core` package, it probably would add more overhead and annoyance to us than value

Reply:
I'm directionally ok with that `export * as Namespace` approach, though I want to make sure that it wouldn't make it harder to do tree shaking or otherwise bloat bundles.

If it is an issue, we might consider an alternative of two package entry points. i.e.

-  In this file, we `export * from "./langchain"` so that there's no namespace wrapper object
- We need to create that file we export from, i.e. `./langchain/index.ts`. It does the `export * from "openai";`
- In `package.json` we add a `exports` for `./langchain`.

So basically I could do either of these as a consumer:

```ts
import { LangChainGithubRepoLoader } from "mongodb-rag-core";
import { LangChainGithubRepoLoader } from "mongodb-rag-core/langchain";
```

Maybe too complex?

Reply:
Interesting thought on the peer dependencies, I'd have to think through it a bit or see it in action

Reply:
i'm notionally in agreement w/ the idea of `mongodb-rag-core/<namespace>`, however in practice i've repeatedly run into issues w/ any packages that we compose like this.

Reply:
that's why i didn't use that pattern. but let me use this as an opportunity to at least properly understand why this doesn't work, even if we cant switch to it..tho maybe we'll be able to switch..let's see.

Reply:
going to merge this PR, and play w/ this in separate PR. would require touching a lot of files b/c so many imports would be changed